homeassistant/util/__init__.py

```diff
@@ -260,25 +285,21 @@ def wrapper(*args, **kwargs):
             else:
                 host = args[0] if args else wrapper
 
-            if not hasattr(host, '_throttle_lock'):
-                host._throttle_lock = threading.Lock()
+            if not hasattr(host, '_throttle'):
+                host._throttle = {}
 
-            if not host._throttle_lock.acquire(False):
-                return None
+            if not id(self) in host._throttle:
+                host._throttle[id(self)] = ThrottleItem()
+            throttle_item = host._throttle[id(self)]
 
-            last_call = getattr(host, '_throttle_last_call', None)
-            # Check if method is never called or no_throttle is given
-            force = not last_call or kwargs.pop('no_throttle', False)
+            force = kwargs.pop('no_throttle', False)
+            if not throttle_item.open(self.min_time, force):
```

Comment:
It is a problem. If we throttle for 5 seconds and the call actually takes 6 seconds - no throttle will be applied.